modules/analytics-objects/src/main/java/com/intuit/wasabi/analyticsobjects/Event.java

```diff
@@ -72,11 +74,16 @@ public Type getType() {
     }
 
     public void setName(Event.Name value) {
+        if (Objects.isNull(value)) {
+            throw new IllegalArgumentException("Event.Name must not be null.");
+        }
         name = value;
-        if (name.toString().equals(IMPRESSION)) {
-            type = Type.IMPRESSION;
-        } else { //todo: for other event types, will need to check this.value
-            type = Type.BINARY_ACTION;
+        switch (name.toString()) {
```

Comment:
It is because an action can be "Like button clicked", "Canceled",... i.e. there can be different actions. We just have no analytics for other than binary actions, thats why we only support those two types. At least that's what I assume.